### DISCUSSION TITLE
Questions about iPXE image build

### DISCUSSION THREAD

**Discussion by wkchong2000** [ipxe/ipxe#368] [Q&A]
I have a few questions that hopefully experts in iPXE can help explain and help me solve my problem. 

1) What's the difference between iPXE.EFI and SNPONLY.EFI image?
2) I built the iPXE image with latest code from github, it is version 1.21.1+, and when I boot up my system (Intel IceLake with Mellanox CX6 100Gbps) card, either the boot failed to get a DHCP IP or it timeout during http connection to chain the image.
When I plug in my network to another DHCP server that's serving iPXE 1.21.1AF, it has no such issue.  What could be the difference between these two images?
3) if I have to build iPXE 1.21.1AF, how should I proceed?

Looking forward to hear from the community.

Thanks
Chris

**Comment by NiKiZe** [comment]
1. Both are EFI builds.
Ipxe contains all drivers that is written natively, SNP is only the SNP/NII driver, the only part is that it only tries to chain via the device that it was booted from. See https://ipxe.org/appnote/buildtargets

2. Use ifstat, post your output

3. https://ipxe.org/download (read the whole page 2-3 times)
